magnum_cluster_api/resources.py

```diff
@@ -1315,6 +1340,82 @@ def get_object(self) -> objects.ClusterClass:
                                 }
                             ],
                         },
+                        {
+                            "name": "ubuntu",
+                            "enabledIf": '{{ if eq .operatingSystem "ubuntu" }}true{{end}}',
+                            "definitions": [
+                                {
+                                    "selector": {
+                                        "apiVersion": objects.KubeadmControlPlaneTemplate.version,
+                                        "kind": objects.KubeadmControlPlaneTemplate.kind,
+                                        "matchResources": {
+                                            "controlPlane": True,
+                                        },
+                                    },
+                                    "jsonPatches": [
+                                        {
+                                            "op": "add",
+                                            "path": "/spec/template/spec/kubeadmConfigSpec/files/-",
+                                            "valueFrom": {
+                                                "template": textwrap.dedent(
+                                                    """\
+                                                    path: "/etc/apt/apt.conf.d/90proxy"
+                                                    owner: "root:root"
+                                                    permissions: "0644"
+                                                    content: "{{ .aptProxyConfig }}"
+                                                    encoding: "base64"
+                                                    """
+                                                ),
+                                            },
+                                        },
+                                        {
+                                            "op": "add",
+                                            "path": "/spec/template/spec/kubeadmConfigSpec/preKubeadmCommands/-",
```

Comment:
```suggestion
                                            "path": "/spec/template/spec/kubeadmConfigSpec/preKubeadmCommands/-",
```

Maybe this shouldn't have the trailing `/-` ?